template_app_config_local.sh

```diff
@@ -0,0 +1,245 @@
+}
+
+prompt_variable() {
+    log_message 9 "# prompt_variable script"
+    declare name=$1
+    declare prompt_text=$2
+    declare is_secret=$3
+    declare default_value=$4
+    if [ "${is_secret}" -eq "1" ]; then
+        read_switch='-sp'
+    else
+        read_switch='-p'
+    fi
+    if [ -n "$default_value" ]; then
+        prompt_text+=" (default: ${default_value})"
+    fi
+    log_message 9 "Prompt variable: <name: $name>; <default_value: $default_value>"
+    if [ -n "${!name}" ]; then
+        log_message 5 "${name}: ${value}"
+    else
+        read ${read_switch} "${prompt_text}: " LOCAL_VARIABLE
+        if [ "${is_secret}" -eq "1" ]; then
+            # Change line, required for usability as the read was made without printing the characters.
+            echo ""
+        fi
+        # log_message 9 "LOCAL_VARIABLE: <${LOCAL_VARIABLE}>"
+        if [ -n "$LOCAL_VARIABLE" ]; then
+            # log_message 9 "LOCAL_VARIABLE defined: <${LOCAL_VARIABLE}>"
+            if [ "${is_secret}" -eq "1" ]; then
+                JINJA2_TEMPLATE_SECRETS+=" -D${name}=${value}"
+            else
+                JINJA2_TEMPLATE_VARIABLES+=" -D${name}=${value}"
+            fi
+        else
+            if [ -n "$default_value" ]; then
+                if [ "${is_secret}" -eq "1" ]; then
+                    JINJA2_TEMPLATE_SECRETS+=" -D${name}=${default_value}"
+                else
+                    JINJA2_TEMPLATE_VARIABLES+=" -D${name}=${default_value}"
+                fi
+            fi
+        fi
+    fi
+}
+
+# Prompt the input variables
+prompt_variables() {
+    log_message 9 "# prompt_variable script"
+    log_message 0 ""
+
+    if [ -n "$JINJA2_TEMPLATE_FILE" ]; then
+        log_message 5 "JINJA2_TEMPLATE_FILE: ${JINJA2_TEMPLATE_FILE}"
+    else
+        read -p 'Template file location (default: ./manifest/app-config.local.yaml.j2): ' JINJA2_TEMPLATE_FILE
+        if [ ! -n "$JINJA2_TEMPLATE_FILE" ]; then
+            JINJA2_TEMPLATE_FILE='./manifest/app-config.local.yaml.j2'
+        fi
+    fi
+
+    if [ -n "$BACKSTAGE_APP_BASE_URL" ]; then
+        log_message 5 "BACKSTAGE_APP_BASE_URL: ${BACKSTAGE_APP_BASE_URL}"
+    else
+        prompt_variable 'BACKSTAGE_APP_BASE_URL' 'Backstage APP Base URL' 0 'localhost:3000'
+    fi
+
+    if [ -n "$BACKSTAGE_BACKEND_BASE_HOST" ]; then
+        log_message 5 "BACKSTAGE_BACKEND_BASE_HOST: ${BACKSTAGE_BACKEND_BASE_HOST}"
+    else
+        prompt_variable 'BACKSTAGE_BACKEND_BASE_HOST' 'Backstage Backend Base Host' 0 'localhost'
+    fi
+
+    if [ -n "$BACKSTAGE_BACKEND_BASE_PORT" ]; then
+        log_message 5 "BACKSTAGE_BACKEND_BASE_PORT: ${BACKSTAGE_BACKEND_BASE_PORT}"
+    else
+        prompt_variable 'BACKSTAGE_BACKEND_BASE_PORT' 'Backstage Backend Base Port' 0 '7007'
+    fi
+
+    if [ -n "$BACKSTAGE_AUTH_SECRET" ]; then
+        log_message 5 "BACKSTAGE_AUTH_SECRET: ********"
+    else
+        prompt_variable 'BACKSTAGE_AUTH_SECRET' 'Backstage Authentication Secret' 1
```

Comment:
The backstage Authentication Secret could be generated OOTB using
 `node -p 'require("crypto").randomBytes(24).toString("base64")'`

!! There is certainly a bash or python equivalent

Reply:
done